### DISCUSSION TITLE
What is adding the nested divs directly #root? (new app)

### DISCUSSION THREAD

**Discussion by hanleybrand** [kaedroho/django-bridge#48] [Q&A] [answered]
I promise I tried googling the answer and looking through the files created via `npm create django-bridge@latest`, but I can't figure out why this index.html:

```html
  <body>
    <div id="root"></div>
  </body>
```
in combination with the default HomeView:

```jsx
  return (
    <>
      <h1>Welcome to your new Django Bridge app!</h1>
    </>
  );
```

renders a header nested in 3 divs below #root?   

```html
<body>
  <div id="root">
    <div>
      <div>
        <div>
            <h1>Welcome to your new Django Bridge app!</h1>
        </div>
      </div>
    </div>
  </div>
</body>
```

**Comment by kaedroho** [comment, marked as answer]
There's a few different places we wrap things in `<div>` tags:

1. https://github.com/django-bridge/django-bridge/blob/c71959e813e5505e2b8c81a79c69210d242a3f1e/packages/react/src/index.tsx#L146

  - This one isn't needed

2. https://github.com/django-bridge/django-bridge/blob/c71959e813e5505e2b8c81a79c69210d242a3f1e/packages/react/src/dirtyform.tsx#L120

  - This one is used to mark the boundary of the root "dirty form scope", which is used to detect any changed forms on the page to prevent navigation away. Another place where there is a "dirty form scope" is inside modals as we want to prevent them from being accidentally closed
  - We need an element here in order to perform `element.querySelector` queries.

3. https://github.com/django-bridge/django-bridge/blob/c71959e813e5505e2b8c81a79c69210d242a3f1e/packages/react/src/components/Browser.tsx#L87

  - If the user navigates to a URL that uses the same view component, the state wouldn't normally be reset by React. So we force this to happen by changing the `key` attribute of this div tag
